metaswitch/common/mib.py

```diff
@@ -267,3 +278,9 @@ def _get_tokenized_mib_details(mib_file, oid):
                 in_braces = not in_braces
 
     return output
+
+
+class DependencyException(Exception):
+    def __init__(self, message, dependency):

```

Comment:
It's worth think about your aims with this class. The main thing you're looking for here is making sure you get sensible output when someone hits an error. I had a go at putting some snippets of this into a python interpreter, and this is what I got out:

```
Traceback (most recent call last):
  File "python", line 6, in <module>
DependencyException
```

Not very useful! So how can we improve this? I have a couple of ideas:

- At the moment you're passing in a message and a dependency. But the message is always the same! Instead, why not make the message part of the name of the exception, leaving you with a single argument. So you could instead call it `MissingDependencyException`, at which point the added "Missing dependency" text is redundant.
- You're inheriting from `Exception` and overriding the `__init__()` method. But the default `__init__()` method is actually very handy here - it takes an argument that it stores off as a string that is printed when the exception is dropped. This seems more useful than encoding extra information that users won't see (and in general it's better to take what you're given rather than rolling your own if you can avoid it).

Merging those two ideas together, you'd end up with the following code to describe your new exception:

```
class MissingDependencyException(Exception):
    pass
```

Much simpler! When I tested this out, I got the following:

```
Traceback (most recent call last):
  File "python", line 4, in <module>
MissingDependencyException: snmptranslate
```
Which is better :). There's an argument that it could give you more information about what to do about this (eg. tell you to go away and install `snmptranslate`). If you want to do more here, come and have a chat with me and we can have a look at it :).

(FWIW, this is one area where Python3 is much better. It has a number of tools that make exception handling more seamless - check [this](https://stackoverflow.com/a/29442282) out!)